File: tests/transpiler/x/pas/bool_chain.pas
{$mode objfpc}
program Main;
uses SysUtils;
function boom(): boolean;
begin
  writeln('boom');
  exit(true);
end;
begin
  writeln(BoolToStr(((1 < 2) and (2 < 3)) and (3 < 4), True));
  writeln(BoolToStr(((1 < 2) and (2 > 3)) and boom(), True));
  writeln(BoolToStr((((1 < 2) and (2 < 3)) and (3 > 4)) and boom(), True));
end.

File: tests/transpiler/x/pl/bool_chain.pl
:- initialization(main).

main :-
    ((1 < 2) , (2 < 3) , (3 < 4) -> writeln(true) ; writeln(false)),
    ((1 < 2) , (2 > 3) , true -> writeln(true) ; writeln(false)),
    ((1 < 2) , (2 < 3) , (3 > 4) , true -> writeln(true) ; writeln(false)).
